### DISCUSSION TITLE
Error accessing record

### DISCUSSION THREAD

**Discussion by longr** [eblondel/zen4R#167] [Q&A]
When I try to access a zenodo record, even the examples, I get an error.

```
> library('zen4R')
> zenodo <- ZenodoManager$new()
> zenodo$getRecordByDOI("10.5281/zenodo.3378733")
Error in if (result$doi == doi) { : argument is of length zero
```

**Comment by eblondel** [comment]
What is your `sessionInfo`? version of zen4R?
I tried with the latest zen4R version, R 4.3.0, and it works

**Reply by longr** [reply]
It is installed through conda

R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Fedora Linux 38 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /home/longr/.conda/envs/test_zen/lib/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] zen4R_0.3

loaded via a namespace (and not attached):
[1] compiler_3.6.1 httr_1.4.0     magrittr_1.5   R6_2.4.0       Rcpp_1.0.1    
[6] xml2_1.2.0     jsonlite_1.8.8 rvest_0.3.3

**Reply by eblondel** [reply]
Ouch!! you have an olddddd version :-) That version can't work with the recent upgrade made by Zenodo. You would need to upgrade

**Reply by eblondel** [reply]
We woul need to see how this could be upgraded https://anaconda.org/r/r-zen4r

**Comment by longr** [comment]
Thanks.  Moved to the latest version and having some issues still.  Might be (probably is) user error.   I have a zenodo link (with embedded token) shared with me:

```
https://zenodo.org/records/127442?token=<token>
```

```
> zenodo <- ZenodoManager$new(token='token=<token>)
> myrec <- zenodo$getRecordByDOI('10.5281/zenodo.12744255')
ℹ Successfully fetched list of published records - page 1
✔ Successfully fetched list of published records!
✔ Successfully fetched record for DOI '10.5281/zenodo.12744255'!
> zen_files <- zenodo$getFiles(myrec$id)
✖ Error while fetching file(s) for record '12744255': Not found.
```

I don't know what I am doing wrong here. Any advice would be appreciated.

**Reply by eblondel** [reply]
Looking at the record, files are under restricted policy:

![image](https://github.com/user-attachments/assets/25985084-8895-4780-a314-f04dd0c6f713)

The `not found` means they are no files available for download. A more explicit message is given when doing `myrec$downloadFiles()`. If you get this message it means you do not have privileges to get these restricted files. If at contrary you have these privileges, then I will need to test if there is something specific to deal with getting restricted files, as it was not tested specifically with zen4R.

**Reply by courtiol** [reply]
Hi @eblondel,

I have bumped into a similar situation: I tried to access restricted data and these data are mine, but I could not access them.

I tried declaring my token, but that does not appear sufficient:
```r
> zenodo <- zen4R::ZenodoManager$new(token = "[my_token]")
> record <- zenodo$getRecordByDOI(doi = "[my doi for restricted data]")
> record$files
named list()
```
I did check that online (after logging into Zenodo) I do have access to the data and I do.

The API seems to allow it since the following works:
```sh
wget --header='Authorization: Bearer [my_token]' https://zenodo.org/api/records/[my doi for restricted data]/files/[my file]/content
```

Thanks for having a look (and possibly implementing that)...
